src/pointer.jl

```diff
@@ -122,3 +122,15 @@ function Base.unsafe_convert(::Type{PtrOrCuPtr{T}}, val) where {T}
     end
     return Base.bitcast(PtrOrCuPtr{T}, ptr)
 end
+
+function memory_type(x::CuPtr)
```

Comment:
Maybe export the enum with `@enum_without_prefix` and cast the value to an instance of it?

Reply:
Sorry, I'm super confused by this comment - are you talking about the return value of this function?

Reply:
Yes, sorry for the dense comment. The return type is now a nondescript integer, right? But we know it should be a value of the CUmemorytype_enum. So it'd be better to return an enum, instead.
However, for users to be able to use that enum, we need to re-export that. That's what the `@enum_without_prefix` macro is for, to remove the redundant CU in `CU_MEMORYTYPE_DEVICE`.